client/src/components/Project/ProjectListing.js

```diff
@@ -0,0 +1,182 @@
+import { deleteProject, clearProjectMsg } from '../../actions/projectAction'
+import { useToasts } from 'react-toast-notifications'
+import Employee from './Project'
+const useStyles = makeStyles(styles)
+const ProjectListing = props => {
+  const { projectData, setPageView } = props
+  const { addToast } = useToasts()
+  const classes = useStyles()
+  const dispatch = useDispatch()
+  const [projectToUpdate, setProjectToUpdate] = useState()
+  const [updateAction, setUpdateAction] = useState()
+  const [showDelDialog, setShowDelDialog] = useState(false)
+  const deleteProjectSuccess = useSelector(
+    state => state.projectReducer.deleteProjectSuccess
+  )
+  const deleteProjectError = useSelector(
+    state => state.projectReducer.deleteProjectError
+  )
+
+  const projectListingHeader = [
+    'Title',
+    'Technology',
+    'Client',
+    'Client Location',
+    'Start Date',
+    'End Date'
+  ]
+
+  useEffect(() => {
+    if (deleteProjectSuccess) {
+      addToast(deleteProjectSuccess, {
+        appearance: 'success',
+        autoDismiss: true
+      })
+      dispatch(clearProjectMsg())
+    }
+  }, [deleteProjectSuccess, addToast, dispatch])
+
+  useEffect(() => {
+    if (deleteProjectError) {
+      addToast(deleteProjectError, {
+        appearance: 'error',
+        autoDismiss: true
+      })
+      dispatch(clearProjectMsg())
+    }
+  }, [deleteProjectError, addToast, dispatch])
+
+  const projectDetails = []
+  let filteredProject
+  if (projectData) {
+    filteredProject = projectData.filter(cls => cls.status !== 'Inactive')
+    filteredProject.map((key, value) => {
+      let {
+        title,
+        technology,
+        client,
+        client_location,
+        startdate,
+        enddate
+      } = key
+
+      const start = new Date(startdate)
+      const end = new Date(enddate)
+      const mnth = ('0' + (start.getMonth() + 1)).slice(-2)
+      const day = ('0' + start.getDate()).slice(-2)
+      startdate = [day, mnth, start.getFullYear()].join('/')
+
+      let formattedEnddate
+      if (enddate) {
+        const mnth = ('0' + (end.getMonth() + 1)).slice(-2)
+        const day = ('0' + end.getDate()).slice(-2)
+        formattedEnddate = [day, mnth, end.getFullYear()].join('/')
+      }
+      enddate = enddate ? formattedEnddate : '-'
+
+      const data = {
+        title,
+        technology,
+        client,
+        client_location,
+        startdate,
+        enddate
+      }
+      projectDetails.push(Object.values(data))
+      return 1
+    })
+    // Get/Delete data from Active projects
+    //  projectData = filteredProject;
+  }
+
+  const links = ['Update', 'Delete']
+
+  const getprojectToUpdate = (projectData, title) => {
+    return projectData.filter(item => {
+      if (item.title === title) return item
+    })
+  }
+
+  const updateUser = val => {
+    setUpdateAction('update')
+    const user = getprojectToUpdate(filteredProject, val[0])
```

Comment:
why dont we pass val[0] at function call instead accessing here?

Reply:
We are retrieving and saving user value to state variable based on Update/delete action.